data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/utils/ServerIdGenerator.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ *  The OpenSearch Contributors require contributions made to
+ *  this file be licensed under the Apache-2.0 license or a
+ *  compatible open source license.
+ */
+
+package org.opensearch.dataprepper.plugins.source.rds.utils;
+
+import java.net.InetAddress;
+import java.util.Random;
+
+public class ServerIdGenerator {
+    static final int MIN_SERVER_ID = 100_000;
+    static final int MAX_SERVER_ID = 999_999;
+
+    public static int generateServerId() {
```

Comment:
Is this value guaranteed to be unique?

Reply:
It's not 100% that it will be unique but the possibility of conflict is very low. 

If there's still server id conflict between two Data Prepper instances, the retry logic provided by StreamScheduler will re-generate a server id and reconnect.